tools/rna_tools/pipmir/pipmir.xml

```diff
@@ -0,0 +1,230 @@
+                    message="No 2bit genomes are available for
+                    the selected input dataset"/>
+                </options>
+            </param>
+            </when>
+            <when value="history">
+                <param name="ownFile" type="data" format="twobit"
+                label="Select the reference genome" />
+            </when>
+        </conditional>
+
+        <!-- optional parameters -->
+        <conditional name="params">
+            <param name="settingsType" type="select"
+                label="Optional parameters"
+                help="You can use the default settings or
+                set custom values for any of pipmir's parameters.">
+              <option value="default">Use defaults</option>
+              <option value="custom">Full parameter list</option>
+            </param>
+            <when value="default" />
+            <!-- Full/advanced params. -->
+            <when value="custom">
+                <param name="min_precursor" type="integer"
+                    value="50" label="Minimum size"
+                    help="Minimum size of a precursor sequence
+                    (Default: 50)"/>
+                <param name="max_precursor" type="integer"
+                    value="500" label="Maximum size"
+                    help="Maximum size of a precursor sequence
+                    (Default: 500)"/>
+                <param name="step_size" type="integer"
+                    value="2" label="Step size"
+                    help="The step size used to identifiy the precursor
+                    from the minimum to the maximum possible
+                    size of precursor (Default: 2)"/>
+                <param name="min_read" type="integer"
+                    value="10" label="Minimum read count"
+                    help="Minimum read count for a mature to be
+                    considered expressed (Default: 10)"/>
+            </when>  <!-- full -->
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="putativeMatures" format="bed"
+        from_work_dir="test_putativeMatures.bed"
+        label="${tool.name} on ${on_string}: putative mature miRNAs"/>
+
+        <data name="predictedPrecursors" format="txt"
+        from_work_dir="test_predictedPrecursors.txt"
+        label="${tool.name} on ${on_string}: predicted precursor"/>
+
+        <data name="predicted_miRNA" format="txt"
+        from_work_dir="test_predicted_miRNAs.txt"
+        label="${tool.name} on ${on_string}: predicted miRNAs"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_bam" value="Aligned.out.sam" ftype="sam" />
+            <param name="genomeSource" value="history" />
+            <param name="ownFile" value="test_seq.2bit" />
+            <param name="settingsType" value="custom" />
+            <param name="step_size" value="10" />
+            <output name="predicted_miRNA" file="test_predicted_miRNAs.txt"
```

Comment:
Can you include the other as well, if they are small ... I have some evil plans how we can make use of these test-data as part of trainings and Galaxy tours.